packages/fxa-auth-server/lib/payments/iap/apple-app-store/app-store-helper.ts

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import {
+  AppStoreServerAPI,
+  Environment,
+  StatusResponse,
+} from 'app-store-server-api';
+import { Container } from 'typedi';
+
+import { AppConfig, AuthLogger } from '../../../types';
+
+export class AppStoreHelper {
+  private log: AuthLogger;
+  private appStoreServerApiClients: {
+    [key: string]: AppStoreServerAPI;
+  };
+  private credentialsByBundleId: any;
+  private environment: Environment;
+
+  constructor() {
+    this.log = Container.get(AuthLogger);
+    const {
+      subscriptions: { appStore },
+    } = Container.get(AppConfig);
+    this.credentialsByBundleId = {};
+    // Initialize App Store Server API client per bundle ID
+    this.environment = appStore.sandbox
+      ? Environment.Sandbox
+      : Environment.Production;
+    this.appStoreServerApiClients = {};
+    for (const [bundleIdWithUnderscores, credentials] of Object.entries(
+      appStore.credentials
+    )) {
+      // Cannot use an actual bundleId (e.g. 'org.mozilla.ios.FirefoxVPN') as the key
+      // due to https://github.com/mozilla/node-convict/issues/250
+      const bundleId = bundleIdWithUnderscores.replace('_', '.');
+      this.credentialsByBundleId[bundleId] = credentials;
+      this.clientByBundleId(bundleId);
+    }
+  }
+
+  /**
+   * Returns an App Store Server API client by bundleId, initializing it first
+   * if needed.
+   */
+  clientByBundleId(bundleId: string): AppStoreServerAPI {
+    if (this.appStoreServerApiClients.hasOwnProperty(bundleId)) {
+      return this.appStoreServerApiClients[bundleId];
+    }
+    const { serverApiKey, serverApiKeyId, issuerId } =
+      this.credentialsByBundleId[bundleId];
```

Comment:
If the bundleId is not in the credentials, this will throw a TypeError and we'll return a 500 error which might not be obvious when debugging on Sentry about why it happened (a mismatch between bundleId defined in Firestore, and the one in auth-server config).

Reply:
Thanks for pointing this out -- I've done the following -- is this what you had in mind for a more descriptive error?

```diff
bdanforth ~/Projects/fxa/packages/fxa-auth-server (fxa-3934-apple-iap-base-class) $ cat ~/fxa-3934-wip.diff 
diff --git a/packages/fxa-auth-server/lib/payments/iap/apple-app-store/app-store-helper.ts b/packages/fxa-auth-server/lib/payments/iap/apple-app-store/app-store-helper.ts
index 9d8ccbc1c1..ed5789f749 100644
--- a/packages/fxa-auth-server/lib/payments/iap/apple-app-store/app-store-helper.ts
+++ b/packages/fxa-auth-server/lib/payments/iap/apple-app-store/app-store-helper.ts
@@ -9,6 +9,7 @@ import {
 import { Container } from 'typedi';
 
 import { AppConfig, AuthLogger } from '../../../types';
+import { AppStoreHelperError } from './types/errors';
 
 export class AppStoreHelper {
   private log: AuthLogger;
@@ -48,6 +49,13 @@ export class AppStoreHelper {
     if (this.appStoreServerApiClients.hasOwnProperty(bundleId)) {
       return this.appStoreServerApiClients[bundleId];
     }
+    if (!this.credentialsByBundleId.hasOwnProperty(bundleId)) {
+      const libraryError = new Error(
+        `No App Store credentials found for app with bundleId: ${bundleId}.`
+      );
+      libraryError.name = AppStoreHelperError.CREDENTIALS_NOT_FOUND;
+      throw libraryError;
+    }
     const { serverApiKey, serverApiKeyId, issuerId } =
       this.credentialsByBundleId[bundleId];
     this.appStoreServerApiClients[bundleId] = new AppStoreServerAPI(
diff --git a/packages/fxa-auth-server/lib/payments/iap/apple-app-store/types/errors.ts b/packages/fxa-auth-server/lib/payments/iap/apple-app-store/types/errors.ts
index f036e122ec..74f905964d 100644
--- a/packages/fxa-auth-server/lib/payments/iap/apple-app-store/types/errors.ts
+++ b/packages/fxa-auth-server/lib/payments/iap/apple-app-store/types/errors.ts
@@ -14,3 +14,7 @@ export enum PurchaseUpdateError {
   INVALID_ORIGINAL_TRANSACTION_ID = 'InvalidOriginalTransactionId',
   OTHER_ERROR = 'OtherError',
 }
+
+export enum AppStoreHelperError {
+  CREDENTIALS_NOT_FOUND = 'CredentialsNotFound',
+}
```